agent/domain/v1/domain.proto

```diff
@@ -0,0 +1,45 @@
+syntax = "proto3";
+package io.spoud.sdm.agent.domain.v1;
+
+option java_multiple_files = true;
+option java_package = "io.spoud.sdm.agent.domain.v1";
+option go_package = "agent.v1";
+
+
+message AgentRef {
+  string agent_username = 1;
+  string transport_path = 2;
+}
+
+message AgentEventType {
+  enum Type{
+    // The agent started (can be useful to see if the agent restarts
+    AGENT_STARTED = 0;
+
```

Comment:
for me those two look kind the same, if started should be available and if ping ok its started. why not "IS_ALIVE" instead?
should we add AGENT_STOP as well?

Reply:
For me it's different. The start is only when the agent starts. This means, if we get a lot of starts, it means that the agent restarts a lot (think of k8s).

PING or IS_ALIVE is like a heartbeat. We could also call it HEARTBEAT.

How do you want to send an AGENT_STOP if the agent is killed? For OOM for example?

Reply:
I though that we could react before shutdown but if not possible nevermind and actually we could use the heartbeat for that, if no response it because its stoped.

Reply:
For a graceful shutdown (upgrade or whatever), yes we can. For a sigkill or a OOMKill no we can't...